xwiki-platform-core/xwiki-platform-help/xwiki-platform-help-ui/src/test/java/org/xwiki/help/XWikiSyntaxMacrosListPageTest.java

```diff
@@ -0,0 +1,177 @@
+        when(queryManagerScriptService.xwql(any())).thenReturn(query);
+        when(query.execute()).thenReturn(List.of("xwiki:XWiki.MyMacro"));
+        
+        // Render the page.
+        Document document = renderHTMLPage(DOCUMENT_REFERENCE);
+        
+        
+        // Assert the values of the cells.
+        Elements trs = document.select("tr");
+        Element headersRow = trs.get(0);
+        Elements headerThs = headersRow.select("th");
+        assertEquals("help.macroList.id", headerThs.get(0).text());
+        assertEquals("help.macroList.name", headerThs.get(1).text());
+        assertEquals("help.macroList.categories", headerThs.get(2).text());
+        assertEquals("help.macroList.description", headerThs.get(3).text());
+        assertEquals("help.macroList.visibility", headerThs.get(4).text());
+        Element myMacroRow = trs.get(1);
+        Elements myMacroRowTds = myMacroRow.select("td");
+        assertEquals("mymacro", myMacroRowTds.get(0).text());
+        assertEquals("/xwiki/bin/view/XWiki/MyMacro", myMacroRowTds.get(0).select("a").attr("href"));
+        assertEquals("My Macro", myMacroRowTds.get(1).text());
+        assertEquals(Set.of("Category1", "Category2"), Arrays.stream(myMacroRowTds.get(2).text().split("\\s*,\\s*"))
+            .collect(Collectors.toSet()));
+        assertEquals("My Macro Description", myMacroRowTds.get(3).text());
+        assertEquals("WIKI", myMacroRowTds.get(4).text());
+        Element velocityMacroRow = trs.get(2);
+        Elements velocityMacroRowTds = velocityMacroRow.select("td");
+        assertEquals("velocity", velocityMacroRowTds.get(0).text());
+        assertEquals("Velocity", velocityMacroRowTds.get(1).text());
+        assertEquals("Development", velocityMacroRowTds.get(2).text());
+        assertEquals("Executes a Velocity script.", velocityMacroRowTds.get(3).text());
+        assertEquals("XWiki.WikiMacroClass_visibility_Global", velocityMacroRowTds.get(4).text());
+        Element htmlMacroRow = trs.get(3);
+        Elements htmlMacroRowTds = htmlMacroRow.select("td");
+        assertEquals("html", htmlMacroRowTds.get(0).text());
+        assertEquals("HTML", htmlMacroRowTds.get(1).text());
+        assertEquals("Development", htmlMacroRowTds.get(2).text());
+        assertEquals("Inserts HTML or XHTML code into the page.", htmlMacroRowTds.get(3).text());
+        assertEquals("XWiki.WikiMacroClass_visibility_Global", htmlMacroRowTds.get(4).text());
+        Element includeMacroRow = trs.get(4);
+        Elements includeMacroRowTds = includeMacroRow.select("td");
+        assertEquals("include", includeMacroRowTds.get(0).text());
+        assertEquals("Include", includeMacroRowTds.get(1).text());
+        assertEquals("Content", includeMacroRowTds.get(2).text());
+        assertEquals("Include other pages into the current page.", includeMacroRowTds.get(3).text());
+        assertEquals("XWiki.WikiMacroClass_visibility_Global", includeMacroRowTds.get(4).text());
```

Comment:
Maybe a more concise/simple solution would be to use XMLUnit to assert this? Something like https://github.com/xwiki/xwiki-platform/blob/1c41e294429b0f8468677593fc289d0c73510b0d/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/UorgsuggestPageTest.java#L124

WDYT?

Reply:
I agree it's more concise but it can also make the test more fragile.